plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsContainerOptionsMapper.groovy

```diff
@@ -0,0 +1,159 @@
+        if ( findOptionWithBooleanValue(options, 'read-only') )
+            containerProperties.setReadonlyRootFilesystem(true);
+    }
+
+    protected static void checkUlimit(CmdLineOptionMap options, ContainerProperties containerProperties) {
+        final ulimits = new ArrayList<Ulimit>()
+        findOptionWithMultipleValues(options, 'ulimit').each { value ->
+            final tokens = value.tokenize('=')
+            final limits = tokens[1].tokenize(':')
+            if ( limits.size() > 1 )
+                ulimits << new Ulimit().withName(tokens[0])
+                        .withSoftLimit(limits[0] as Integer).withHardLimit(limits[1] as Integer)
+            else
+                ulimits << new Ulimit().withName(tokens[0]).withSoftLimit(limits[0] as Integer)
+        }
+        if ( ulimits.size() > 0 )
+            containerProperties.setUlimits(ulimits)
+    }
+
+    protected static LinuxParameters checkLinuxParameters(CmdLineOptionMap options) {
+        final params = new LinuxParameters()
+        boolean atLeastOneSet = false
+
+        // shared Memory Size
+        def value = findOptionWithSingleValue(options, 'shm-size')
+        if ( value && value.length() > 0 ) {
+            params.setSharedMemorySize(value as Integer)
+            atLeastOneSet = true
+        }
+
+        // tmpfs mounts, e.g --tmpfs /run:rw,noexec,nosuid,size=64
+        final tmpfs = new ArrayList<Tmpfs>()
+        findOptionWithMultipleValues(options, 'tmpfs').each { ovalue ->
+            def matcher = ovalue =~ /^(?<path>.*):(?<options>.*?),size=(?<sizeMiB>.*)$/
+            if (matcher.matches()) {
+                tmpfs << new Tmpfs().withContainerPath(matcher.group('path'))
+                        .withSize(matcher.group('sizeMiB') as Integer)
+                        .withMountOptions(matcher.group('options').tokenize(','))
+            } else {
+                throw new IllegalArgumentException("Found a malformed value '${ovalue}' for --tmpfs option")
+            }
+        }
+        if ( tmpfs.size() > 0 ) {
+            params.setTmpfs(tmpfs)
+            atLeastOneSet = true
+        }
+
+        // swap limit equal to memory plus swap
+        value = findOptionWithSingleValue(options, 'memory-swap')
+        if ( value && value.length() > 0) {
+            params.setMaxSwap(value as Integer)
+            atLeastOneSet = true
+        }
+
+        // run an init inside the container
+        if ( findOptionWithBooleanValue(options, 'init') ) {
+            params.setInitProcessEnabled(true)
+            atLeastOneSet = true
+        }
+
+        // tune container memory swappiness
+        value = findOptionWithSingleValue(options, 'memory-swappiness')
+        if ( value && value.length() > 0 ) {
+            params.setSwappiness(value as Integer)
+            atLeastOneSet = true
+        }
+
+        return atLeastOneSet ? params : null
+    }
+
```

Comment:
It should be added a unit test each of the option

Reply:
done, see https://github.com/nextflow-io/nextflow/pull/2471/commits/d1d9da011fb3e422b08ebc9b3352768f25605984